rt/frontend/jaxws/src/main/java/org/apache/cxf/jaxws/spi/WrapperClassNamingConvention.java

```diff
@@ -0,0 +1,148 @@
+     * <th>SEI</th><th>anonymous</th>
+     * <th>{@code getWrapperClassPackageName()} return value</th>
+     * </tr>
+     * <tr>
+     * <td>{@code org.example.Service}</td>
+     * <td>{@code false}</td>
+     * <td>{@code org.example.jaxws_asm.service}</td>
+     * </tr>
+     * <tr>
+     * <td>{@code org.example.OuterClass$Service}</td>
+     * <td>{@code false}</td>
+     * <td>{@code org.example.jaxws_asm.outerclass_service}</td>
+     * </tr>
+     * <tr>
+     * <td>{@code org.example.Service}</td>
+     * <td>{@code true}</td>
+     * <td>{@code org.example.jaxws_asm_an.service}</td>
+     * </tr>
+     * </table>
+     *
+     * @since 4.2.0
+     */
+    class DefaultWrapperClassNamingConvention implements WrapperClassNamingConvention {
+        public static final String DEFAULT_PACKAGE_NAME = "defaultnamespace";
+        private static final Pattern JAVA_PACKAGE_NAME_SANITIZER_PATTERN = Pattern.compile("[^a-zA-Z0-9_]");
+
+        @Override
+        public String getWrapperClassPackageName(Class<?> sei, boolean anonymous) {
+            final String className = sei.getName();
+            final int start = className.startsWith("[L") ? 2 : 0;
+            final int end = className.lastIndexOf('.');
+            final String pkg;
+            final String cl;
+            if (end >= 0) {
+                pkg = className.substring(start, end);
+                cl = className.substring(end + 1);
+            } else {
+                pkg = DefaultWrapperClassNamingConvention.DEFAULT_PACKAGE_NAME;
+                cl = className;
+            }
+
+            return pkg
+                    + (anonymous ? ".jaxws_asm_an." : ".jaxws_asm.")
+                    + JAVA_PACKAGE_NAME_SANITIZER_PATTERN.matcher(cl).replaceAll("_").toLowerCase(Locale.ROOT);
+        }
+
+    }
+
+    /**
+     * An implementation restoring the behavior of CXF before version 4.2.0.
+     * <p>
+     * Unlike with {@link DefaultWrapperClassNamingConvention}, this implementation's
+     * {@link #getWrapperClassPackageName(Class, boolean)} takes only package name
+     * of the given {@code sei} into account.
+     * Therefore naming clashes may occur if two SEIs are in the same package
+     * and both of them have a method with the same name but possibly different signature.
+     * <p>
+     * Examples:
+     * <table>
+     * <tr>
+     * <th>SEI</th>
+     * <th>anonymous</th>
+     * <th>{@code getWrapperClassPackageName()} return value</th>
+     * </tr>
+     * <tr>
+     * <td>{@code org.example.Service}</td>
+     * <td>{@code false}</td>
+     * <td>{@code org.example.jaxws_asm}</td>
+     * </tr>
+     * <tr>
+     * <td>{@code org.example.OuterClass$Service}</td>
+     * <td>{@code false}</td>
+     * <td>{@code org.example.jaxws_asm}</td>
+     * </tr>
+     * <tr>
+     * <td>{@code org.example.Service}</td>
+     * <td>{@code true}</td>
+     * <td>{@code org.example.jaxws_asm_an}</td>
+     * </tr>
+     * </table>
+     *
+     * @since 4.2.0
```

Comment:
Super nit, I think we could target it to 4.1.1
```suggestion
     * @since 4.1.1
```

Reply:
Addressed in adc8f6f